rfcs/0023-musl-libc.md

```diff
@@ -0,0 +1,158 @@
+by a number of important projects you may be familiar with
+large userbases, including:
+* Alpine Linux - [#70 on Distrowatch](https://distrowatch.com/table.php?distribution=alpine) but very popular amongst docker users for producing slim container images.
+* [OpenWRT/LEDE](https://openwrt.org/) - #1 open-source Linux router firmware project; foundation of most other projects targetting routers.
+More projects and details of how they use musl can be found here:
+
+https://wiki.musl-libc.org/projects-using-musl.html
+
+## Why Nixpkgs?
+
+The importance of musl is not the primary point of contention in this RFC,
+instead perhaps the main question is whether such support belongs in Nixpkgs or not.
+
+The main arguments for inclusion are:
+* convenience of use
+* foundation for exciting future work: musl is widely used by high-level languages
+  as the libc implementation used to produce statically linked programs:
+  these are easy to deploy, launch quickly, and only include the code required.
+  (NOTE: currently musl support prefers dynamic linking and shared libraries
+   as is the strong preference in Nixpkgs)
+* Software sometimes must be patched to compile or run with musl; in @dtzWill's experience,
+  these changes are largely fixes improving compliance and correctness resulting in
+  higher-quality programs.  Recent versions of glibc have started taking stronger stances
+  on enforcing compliance (look at patch fallout folllowing any glibc upgrade in last year or so)
+  resulting in overlapping work from both sides.
+
+And to a large extent:
+"Why not?" -- similar to including support for architectures such as Aarch64 or RISC-V,
+  and just like support for those architectures it's relatively clear that pushing them
+  into private forks would be detrimental to the nixpkgs project as well as all users
+  interested in using Nixpkgs on those platforms/architectures.
+
+musl is clearly useful for a variety of important use cases,
+however including support has a few costs (see Drawbacks, below):
+do folks believe the costs are too high?
+
+
+## Additional Resources
+
+* [musl FAQ](https://www.musl-libc.org/faq.html)
+* [projects using musl](https://wiki.musl-libc.org/projects-using-musl.html)
+* [Slides from a talk discussing various libcs, 2014](http://events17.linuxfoundation.org/sites/events/files/slides/libc-talk.pdf)
+
+## Related Isssues
+
+* [big musl PR](https://github.com/NixOS/nixpkgs/pull/34645)
+* [issues matching "musl", newest first](https://github.com/NixOS/nixpkgs/search?o=desc&q=musl&s=created&type=Issues&utf8=%E2%9C%93)
+* [2015 libc discussion](https://github.com/NixOS/nixpkgs/issues/6221#issuecomment-116754223)
+
+# Detailed design
+[design]: #detailed-design
+
+I believe the linked implementation from current nixpkgs master is perhaps
+the best embodiment of "Detailed Design"?
+
+Linky: [bd7d5b365799a145717d31122ebfd24b51fea117](https://github.com/NixOS/nixpkgs/commit/bd7d5b365799a145717d31122ebfd24b51fea117)
```

Comment:
I suspect this link is wrong :)

Reply:
I think it's right, it's just revision of master at time of writing.

Reply:
Maybe a diff or merge commit would be better?